### PR TITLE
Api tests part 2

### PR DESCRIPTION
**Pull request by aboedo**
Part 2 of API tests. 

Adds API tests for all of our public APIs. 

This is also already highlighting a lot of problems in our Unity API that we should fix (as a follow-up PR):
- the casing is inconsistent (all uppercase for properties except for Discount)
- some properties are read/write, but shouldn't be (again, Discount)

### CODE REVIEW

**Comment by aboedo** [comment]
tests will fail (they pass locally) until #103 is merged and this gets rebased from it

File: IntegrationTests/Assets/PurchasesAPITests.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        purchases.appUserID = "abc";
+        purchases.productIdentifiers = new[]
+        {
+            "a", "b", "c"
+        };
+
+        purchases.listener = new CustomListener();
+        purchases.observerMode = true;
+        purchases.userDefaultsSuiteName = "suitename";
+        purchases.proxyURL = "https://proxy-url.revenuecat.com";
```
Comment (vegaro) [inline IntegrationTests/Assets/PurchasesAPITests.cs:33]: There are some lowercase properties here too

File: IntegrationTests/Assets/AttributionNetworkAPITests.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        Purchases.AttributionNetwork network = Purchases.AttributionNetwork.ADJUST;
+        switch (network)
+        {
+            case Purchases.AttributionNetwork.ADJUST:
+            case Purchases.AttributionNetwork.BRANCH:
+            case Purchases.AttributionNetwork.TENJIN:
+            case Purchases.AttributionNetwork.FACEBOOK:
+            case Purchases.AttributionNetwork.APPSFLYER:
+            case Purchases.AttributionNetwork.APPLE_SEARCH_ADS:
```
[the remark this answers is not in the document]
Comment (aboedo) [reply IntegrationTests/Assets/AttributionNetworkAPITests.cs:16]: great catch. I added an obsoletion in the next PR #112

File: IntegrationTests/Assets/PurchasesAPITests.cs (hunk 1 above)
Comment (aboedo) [reply IntegrationTests/Assets/PurchasesAPITests.cs:33]: another great catch, also updated in #112

File: IntegrationTests/Assets/StoreTransactionAPITests.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+using UnityEngine;
+
+namespace DefaultNamespace
+{
+    public class StoreTransactionAPITests : MonoBehaviour
+    {
+        private void Start()
+        {
+            // todo: update to read-only
+            // also, should we include the new properties?
```
[the remark this answers is not in the document]
Comment (aboedo) [reply IntegrationTests/Assets/StoreTransactionAPITests.cs:11]: the rest of the stuff on `StoreTransaction`, but we'd have to untangle the Storekit transaction vs backend transaction mess first

**Pull request merged**